### PR TITLE
Refactor garden-project chart into Golang component

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity quality robustness
/kind technical-debt test
/priority normal
/topology seed
/platform all
/exp intermediate

**What this PR does / why we need it**:
This PR refactors the reconciliation of the RBAC resources for a `Project` and adopts the component concept. Consequently, it eliminates the Helm chart and adds full unit tests, instead. This improves several aspects and the overall code quality.

**Which issue(s) this PR fixes**:
Part of #2754

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by timebertt — DISMISSED** [review]
/lgtm

**Comment by timebertt** [comment]
Should we merge this one, or does anyone else want to have a look?

File: pkg/operation/botanist/projectrbac/projectrbac.go [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+	NameProjectMember = "gardener.cloud:system:project-member"
+	NameProjectViewer = "gardener.cloud:system:project-viewer"
+)
+
+type Interface interface {
+	component.Deployer
+	DeleteStaleExtensionRolesResources(context.Context) error
+}
+
+// New creates a new instance of DeployWaiter for the kube-scheduler.
```
Comment (stoyanr) [inline pkg/operation/botanist/projectrbac/projectrbac.go]: This comment seems wrong.

**Review by stoyanr — DISMISSED** [review]

File: pkg/operation/botanist/projectrbac/projectrbac.go (hunk 1 above)
Comment (rfranzke) [reply pkg/operation/botanist/projectrbac/projectrbac.go]: Fixed, thanks!

**Comment by rfranzke** [comment]
Rebased and fixed https://github.com/gardener/gardener/pull/3458#discussion_r567809561.

**Review by timebertt — APPROVED** [review]
/lgtm

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**